src/utils/assetFormatters.ts

```diff
@@ -0,0 +1,83 @@
+import { Chain, Asset } from "./types";
+import { amountToMainUnit, formatAmount } from "./helper";
+import { getTokenInfo } from "~/api/adamik/token";
+import { getChains } from "~/api/adamik/chains";
+
+// Use a subset of Asset properties that we need
+type AssetIdentifier = Pick<Asset, "chainId" | "isToken" | "assetId">;
+
+export interface FormatAssetAmountOptions {
+  asset: AssetIdentifier;
```

Comment:
Just naming, maybe more explicit and consistent
```suggestion
  assetInfo: AssetFormatInfo;
```